File: org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/basic/table/TableRowTileMapping.java
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.scout.rt.client.ui.basic.table;

import org.eclipse.scout.rt.client.ui.tile.ITile;
import org.eclipse.scout.rt.platform.Bean;
import org.eclipse.scout.rt.platform.reflect.AbstractPropertyObserver;

@Bean
public class TableRowTileMapping extends AbstractPropertyObserver implements ITableRowTileMapping {

  @Override
  public ITableRow getTableRow() {
    return (ITableRow) propertySupport.getProperty(PROP_TABLE_ROW);
  }

  @Override
  public TableRowTileMapping withTableRow(ITableRow tableRow) {
    propertySupport.setProperty(PROP_TABLE_ROW, tableRow);
    return this;
  }

  @Override
  public ITile getTile() {
    return (ITile) propertySupport.getProperty(PROP_TILE);
  }

  @Override
  public TableRowTileMapping withTile(ITile tile) {
    propertySupport.setProperty(PROP_TILE, tile);
    return this;
  }
}

File: org.eclipse.scout.rt.ui.html/src/main/js/scout/table/TableRowTileMapping.js
/*
 * Copyright (c) 2014-2018 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
scout.TableRowTileMapping = function() {
  scout.TableRowTileMapping.parent.call(this);
  this.tableRow = null;
  this.tile = null;
  this._addWidgetProperties(['tile']);
};
scout.inherits(scout.TableRowTileMapping, scout.Widget);
